lib/index.js

```diff
@@ -52,6 +52,27 @@ internals.isFunction = function(functionToCheck) {
   );
 };
 
+internals.getKeys = async function(decoded, options) {
+  // if keyFunc is function allow dynamic key lookup: https://git.io/vXjvY
+  const { key, ...extraInfo } = internals.isFunction(options.key)
```

Comment:
Note: using `...extraInfo` will break node.js backward compatibility and will require a new _major_ version.
I'm _fine_ with this as _most_ people are using LTS 8 or even 10. 
Just saying.

Reply:
This was already there, not sure when it was introduced, so I guess nobody should experience a breakage who haven't already experienced this already.